src/components/Comment/index.tsx

```diff
@@ -225,26 +243,65 @@ export class CommentBase extends React.Component<Props, State> {
     );
   }
 
+  onEdit = () => {
+    const {
+      commentId,
+      dispatch,
+      fileName,
+      line,
+      initialComment,
+      versionId,
+    } = this.props;
+    dispatch(
+      commentsActions.beginComment({
+        commentId: commentId || undefined,
+        fileName,
+        line,
+        versionId,
+      }),
+    );
+    if (initialComment && initialComment.content) {
+      this.setState({ commentText: initialComment.content });
```

Comment:
What is this needed for? When I remove it, nothing in the UI seems to break. 

Mixing `setState` and Redux breaks things like time travel so this would need to be moved into the reducer if it's still needed. The `initialComment` text is already [initialized](https://github.com/mozilla/addons-code-manager/blob/5c957fe5cc75afd87d8102b9d966c159c6443e38/src/components/Comment/index.tsx#L65) during comment construction so maybe it's not necessary to do set it again? When the user types text into the box, the state also gets updated.

If this really is needed, there is [`pendingCommentText`](https://github.com/mozilla/addons-code-manager/blob/67542ab1af136f32cd8e37f72be74a28540152c2/src/reducers/comments.tsx#L72) which achieves something similar.

Reply:
Without this, the comment text in text area will not be properly set when editing the comment for the second time. STR:
1. Click "Edit" button
2. Modify the text
3. Click "Discard", then "Confirm".
4. Click "Edit" again. The text in the text area is the discarded comment. It is expected to be original comment.

I thought about using `pendingCommentText`, but still did not figure out a proper way to set text of text area, so I temporarily removed this part in the updated patch.

Reply:
Filed #1287